extensions/panache/panache-common/deployment/src/main/java/io/quarkus/panache/common/deployment/JandexUtil.java

```diff
@@ -308,6 +314,42 @@ public static int getLoadOpcode(Type jandexType) {
         return Opcodes.ALOAD;
     }
 
+    public static ClassInfo getEnclosingClass(AnnotationInstance annotationInstance) {
+        switch (annotationInstance.target().kind()) {
+            case FIELD:
+                return annotationInstance.target().asField().declaringClass();
+            case METHOD:
+                return annotationInstance.target().asMethod().declaringClass();
+            case METHOD_PARAMETER:
+                return annotationInstance.target().asMethodParameter().method().declaringClass();
+            case CLASS:
+                return annotationInstance.target().asClass();
+            case TYPE:
+                return annotationInstance.target().asType().asClass(); // TODO is it legal here or should I throw ?
+            default:
+                throw new RuntimeException(); // this should not occurs
+        }
+    }
+
+    public static boolean isSubclassOf(IndexView index, ClassInfo info, DotName parentName) throws BuildException {
+        if (info.superName().equals(DOTNAME_OBJECT)) {
+            return false;
+        }
+        if (info.superName().equals(parentName)) {
+            return true;
+        }
+
+        // climb up the hierarchy of classes
+        Type superType = info.superClassType();
+        ClassInfo superClass = index.getClassByName(superType.name());
+        if (superClass == null) {
+            // this can happens if the parent is not inside the Yandex index
```

Comment:
```suggestion
            // this can happens if the parent is not inside the Jandex index
```

Reply:
Ah funny, I always pronounce Jandex Yandex and this time I wrote it ;)

Yandex is the equivalent of Google in Russia ... integrated once in a website for Russia Maps ...

Reply:
Fixed